File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        var product = 1;
        var zeroCount = 0;
        
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                zeroCount++;
            } else {
                product *= nums[i];
            }
        }

        var result = new int[nums.length];

        for (int i = 0; i < nums.length; i++) {
            if (zeroCount > 1) {
                result[i] = 0;
            } else if (nums[i] == 0) {
                result[i] = product;
            } else if (zeroCount == 1) {
                result[i] = 0;
            } else {
                result[i] = product / nums[i];
            }
        }

        return result;
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-1.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        result = []
        queue = deque()

        for i in range(k):
            while len(queue) > 0 and nums[i] > queue[-1]:
                queue.pop()
            queue.append(nums[i])
        
        result.append(queue[0])

        for i in range(k, len(nums)):
            if nums[i - k] == queue[0]:
                queue.popleft()
            while len(queue) > 0 and nums[i] > queue[-1]:
                queue.pop()
            queue.append(nums[i])
            result.append(queue[0])
        
        return result
        